shark-graph/src/main/java/shark/HeapObject.kt

```diff
@@ -386,13 +386,11 @@ sealed class HeapObject {
      * This may trigger IO reads.
      */
     fun readFields(): Sequence<HeapField> {
-      val fieldReader by lazy {
-        hprofGraph.createFieldValuesReader(readRecord())
-      }
+      val fieldReader = hprofGraph.createFieldValuesReader(readRecord())
       return instanceClass.classHierarchy
           .map { heapClass ->
             heapClass.readRecord()
-                .fields.asSequence()
+                .fields
```

Comment:
![image](https://user-images.githubusercontent.com/2152942/74276409-4ce75800-4cdb-11ea-82f0-27c873d9846e.png)

`invoke()` itself is taking a lot of time. Decompiled to Java equivalent looked like this (without lazy):
![image](https://user-images.githubusercontent.com/2152942/74276930-49080580-4cdc-11ea-995e-e4511c17fbc4.png)

Since it's basically a loop inside loop it is creating instances of Functions inside inner loop and that one seems to be expensive? I am not sure I got it completely right..